cmd/integration-test-server/main.go

```diff
@@ -399,14 +399,14 @@ func checkInvalidSamplesError() (int, error) {
 		} else if regexp.MustCompile(`Failed to submit mined result[\s\S]+invalid samples`).MatchString(logText) {
 			for kvIdx, block := range legacyKVs {
 				if block != "" && strings.Contains(logText, block) {
-					log.Warn("By design error", "block", block, "kvIdx", kvIdx, "error", "invalid samples")
+					log.Warn("Known error", "block", block, "kvIdx", kvIdx, "error", "invalid samples")
 					delete(legacyKVs, kvIdx)
 					count++
 					continue
 				}
 				delete(minedEmptyKVs, kvIdx)
 			}
-		} else if strings.Contains(logText, "Mining result loop get result") {
+		} else if strings.Contains(logText, "Submit mined result done") {
```

Comment:
Could you please provide more explanation about why the previous pattern failed to capture the issue and how the changed pattern addresses it?